### PR TITLE
Fix memory allocation and GUI update outside of main thread (macOS compatibility)

### PR DESCRIPTION
**Pull request by belotv**
Fix macOS compatibility by:
- Allocating properly large array (stack allocation not permitted for array holding large response like image)
- Use default browser to open internet link on macOS
- Move button toggles to main thread as MacOS only allows GUI update from main thread

### CODE REVIEW

File: src/lxi_gui-window.c [hunk 1]
```diff
@@ -421,6 +426,18 @@ static void vxi11_device(const char *address, const char *id)
   list_add_instrument(self_global, address, id);
 }
 
+static gboolean
+gui_update_script_run_worker_function_finished_thread(gpointer data)
+{
+  LxiGuiWindow *self = data;
+
+  // Restore search button
+    gtk_toggle_button_set_active(self->toggle_button_script_run, false);
```
Comment (lundmar) [inline src/lxi_gui-window.c:435]: Little bit too much indentation here.

File: src/lxi_gui-window.c [hunk 2]
```diff
@@ -865,8 +894,7 @@ send_worker_thread(gpointer data)
 error_no_input:
   // Restore send button state
   // Defer!
-  gtk_widget_set_sensitive(GTK_WIDGET(self->toggle_button_scpi_send), true);
-  gtk_toggle_button_set_active(self->toggle_button_scpi_send, false);
+    g_idle_add(gui_update_send_worker_finished_thread, self);
```
Comment (lundmar) [inline src/lxi_gui-window.c:898]: Indentation..

File: src/lxi_gui-window.c [hunk 3]
```diff
@@ -2207,8 +2236,7 @@ script_run_worker_function(gpointer data)
   lua_close(L);
 
   // Restore script run button
-  gtk_widget_set_sensitive(GTK_WIDGET(self->toggle_button_script_run), true);
-  gtk_toggle_button_set_active(self->toggle_button_script_run, false);
+    g_idle_add(gui_update_script_run_worker_function_finished_thread, self);
```
Comment (lundmar) [inline src/lxi_gui-window.c:2240]: Indentation..

File: src/scpi.c [hunk 4]
```diff
@@ -105,23 +105,24 @@ int scpi(char *ip, int port, int timeout, lxi_protocol_t protocol, char *command
...
-
+    free(response);
     return 0;
 
 error_send:
 error_receive:
 
     // Disconnect
     lxi_disconnect(device);
+    free(response);
```
Comment (lundmar) [inline src/scpi.c:116]: Remove this free() else we get a double free error.

**Comment by lundmar** [comment]
All the screenshot plugins are now leaking in case of error but you don't need to fix those. I can take care of them. If you fix the stuff I mentioned we are good to merge :)

File: src/scpi.c (hunk 4 above)
Comment (belotv) [reply src/scpi.c:116]: Indeed missed the fact that each error goto did not have a return, will fix that.

**Comment by belotv** [comment]
I fixed the double free however for indent I used the GNU indent command line tool, which does not seem to have the same indent rules as your files. Which IDE or tool do you use for indentation ?

**Comment by lundmar** [comment]
Oh no, please do not run it through an indent tool! :)

They rarely do a good job - they will almost always ruin the intended look of the author. Many of these tools are especially bad at breaking up long lined function prototypes and other stuff. I think your output proves my point :)

I use vim to automatically indent my functions as I go and fix up whatever mistakes it makes. For lxi-tools I'm using 2 space indentation, no tabs - same as the GTK project. I used to use 3 space notation and I plan to fix all the files but now is not the time to do so.

**Pull request merged**